[BUG] - FreeBSD Support Broken

## Description

Compiling the package on FreeBSD from cargo currently fails due to broken support in keyring-rs (see hwchen/keyring-rs#36).
In addition, using the installer as listed in the readme for the project also fails.

## Steps to reproduce

Attempt to install termscp using the installer with
`curl --proto '=https' --tlsv1.2 -sSLf "https://git.io/JBhDb" | sh`.

Once that fails, attempt to install from cargo directly using:
`cargo install --locked termscp`

Removing the `--locked` flag does not help.

## Expected behaviour

Normal installation should occur.

## Environment

- OS: FreeBSD 13.1-RELEASE
- Architecture x86_64
- Rust version: 1.67.0 (fc594f156 2023-01-24)

## Log

Attempting to use the installer using `curl --proto '=https' --tlsv1.2 -sSLf "https://git.io/JBhDb" | sh` fails with this:
```
sh: !: not found
```

Attempting to install termscp with cargo using `cargo install --locked termscp` fails with this:
```
error[E0583]: file not found for module `platform`
  --> /.cargo/registry/src/github.com-1ecc6299db9ec823/keyring-1.2.0/src/lib.rs:79:1
   |
79 | mod platform;
   | ^^^^^^^^^^^^^
   |
   = help: to create the module `platform`, create file "/.cargo/registry/src/github.com-1ecc6299db9ec823/keyring-1.2.0/src/platform.rs" or "/.cargo/registry/src/github.com-1ecc6299db9ec823/keyring-1.2.0/src/platform/mod.rs"
error[E0412]: cannot find type `Error` in module `crate::platform`
  --> /.cargo/registry/src/github.com-1ecc6299db9ec823/keyring-1.2.0/src/error.rs:15:38
   |
15 |     PlatformFailure(crate::platform::Error),
   |                                      ^^^^^ not found in `crate::platform`
   |
help: consider importing one of these items
   |
7  | use core::error::Error;
   |
7  | use core::fmt::Error;
   |
7  | use crate::Error;
   |
7  | use std::error::Error;
   |
     and 2 other candidates
help: if you import `Error`, refer to it directly
   |
15 -     PlatformFailure(crate::platform::Error),
15 +     PlatformFailure(Error),
   |
error[E0412]: cannot find type `Error` in module `crate::platform`
  --> /.cargo/registry/src/github.com-1ecc6299db9ec823/keyring-1.2.0/src/error.rs:21:38
   |
21 |     NoStorageAccess(crate::platform::Error),
   |                                      ^^^^^ not found in `crate::platform`
   |
help: consider importing one of these items
   |
7  | use core::error::Error;
   |
7  | use core::fmt::Error;
   |
7  | use crate::Error;
   |
7  | use std::error::Error;
   |
     and 2 other candidates
help: if you import `Error`, refer to it directly
   |
21 -     NoStorageAccess(crate::platform::Error),
21 +     NoStorageAccess(Error),
   |
error[E0425]: cannot find function `platform` in module `platform`
  --> /.cargo/registry/src/github.com-1ecc6299db9ec823/keyring-1.2.0/src/lib.rs:71:15
   |
71 |     platform::platform()
   |               ^^^^^^^^ not found in `platform`
error[E0425]: cannot find function `set_password` in module `platform`
   --> /.cargo/registry/src/github.com-1ecc6299db9ec823/keyring-1.2.0/src/lib.rs:140:19
    |
140 |         platform::set_password(&self.target, password)
    |                   ^^^^^^^^^^^^ not found in `platform`
error[E0425]: cannot find function `get_password` in module `platform`
   --> /.cargo/registry/src/github.com-1ecc6299db9ec823/keyring-1.2.0/src/lib.rs:148:19
    |
148 |         platform::get_password(&mut map)
    |                   ^^^^^^^^^^^^ not found in `platform`
error[E0425]: cannot find function `get_password` in module `platform`
   --> /.cargo/registry/src/github.com-1ecc6299db9ec823/keyring-1.2.0/src/lib.rs:158:34
    |
158 |         let password = platform::get_password(&mut map)?;
    |                                  ^^^^^^^^^^^^ not found in `platform`
error[E0425]: cannot find function `delete_password` in module `platform`
   --> /.cargo/registry/src/github.com-1ecc6299db9ec823/keyring-1.2.0/src/lib.rs:166:19
    |
166 |         platform::delete_password(&self.target)
    |                   ^^^^^^^^^^^^^^^ not found in `platform`
Some errors have detailed explanations: E0412, E0425, E0583.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `keyring` due to 8 previous errors
```

## Additional information

I am not sure if keyring-rs can be avoided until patched on FreeBSD or if this issue is currently purely on keyring-rs, but perhaps the installer could be fixed.

--- a/install.sh
+++ b/install.sh
@@ -13,6 +13,8 @@ GITHUB_URL="https://github.com/veeso/termscp/releases/download/v${TERMSCP_VERSIO
 DEB_URL="${GITHUB_URL}/termscp_${TERMSCP_VERSION}_amd64.deb"
 RPM_URL="${GITHUB_URL}/termscp-${TERMSCP_VERSION}-1.x86_64.rpm"
 
+PATH="$PATH:/usr/sbin"
+
 set -eu
 printf "\n"
 
@@ -92,16 +94,20 @@ test_writeable() {
 }
 
 elevate_priv() {
-    if ! has sudo; then
-        error 'Could not find the command "sudo", needed to install termscp on your system.'
-        info "If you are on Windows, please run your shell as an administrator, then"
-        info "rerun this script. Otherwise, please run this script as root, or install"
-        info "sudo."
-        exit 1
-    fi
-    if ! sudo -v; then
+    if has sudo; then
+      if ! sudo -v; then
         error "Superuser not granted, aborting installation"
         exit 1
+      fi
+      sudo="sudo"
+    elif has doas; then
+      sudo="doas"
+    else
+      error 'Could not find the commands "sudo" or "doas", needed to install termscp on your system.'
+      info "If you are on Windows, please run your shell as an administrator, then"
+      info "rerun this script. Otherwise, please run this script as root, or install"
+      info "sudo or doas."
+      exit 1
     fi
 }
 
@@ -110,9 +116,7 @@ elevate_priv_ex() {
     if test_writeable "$check_dir"; then
         sudo=""
     else
-        warn "Root permissions are required to install dependecies"
         elevate_priv
-        sudo="sudo"
     fi
     echo $sudo
 }

